<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Window</class>
 <widget class="QWidget" name="Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>802</width>
    <height>591</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Algorithme de SOM</string>
  </property>
  <widget class="QGroupBox" name="Interface">
   <property name="geometry">
    <rect>
     <x>-10</x>
     <y>-20</y>
     <width>300</width>
     <height>620</height>
    </rect>
   </property>
   <property name="title">
    <string>GroupBox</string>
   </property>
   <widget class="QPushButton" name="NouveauRes">
    <property name="geometry">
     <rect>
      <x>85</x>
      <y>50</y>
      <width>130</width>
      <height>37</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string>Nouveau réseau</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="ExecutionGrp">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>291</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QProgressBar" name="ProgressBar">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>285</width>
       <height>23</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="textVisible">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="StartBtn">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Lancer apprentissage</string>
     </property>
    </widget>
    <widget class="QPushButton" name="PauseBtn">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>10</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Pause apprentissage</string>
     </property>
    </widget>
    <widget class="QLabel" name="NbrIterations">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>171</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Itérations : </string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="Titre1">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>120</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Taille du réseau</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
    <property name="wordWrap">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Lignes">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Lignes :</string>
    </property>
   </widget>
   <widget class="QLabel" name="Colonnes">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Colonnes :</string>
    </property>
   </widget>
   <widget class="Line" name="Line1">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>271</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="ColValue">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>180</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="ErreurLignes">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>150</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSpinBox" name="LigneValue">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>150</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="ErreurColonnes">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>180</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="ErreurBeta">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>300</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="Parametres">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>291</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="Titre2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>10</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Paramètres du réseau</string>
     </property>
    </widget>
    <widget class="QSlider" name="AlphaSlider">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>181</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>1000</number>
     </property>
     <property name="sliderPosition">
      <number>1000</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QRadioButton" name="Distance1">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="tabletTracking">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Distance Euclidienne</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSpinBox" name="BetaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>120</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="Line" name="Line2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>271</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="AlphaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>50</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="Alpha">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>47</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Alpha :</string>
     </property>
    </widget>
    <widget class="QLabel" name="Beta">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>47</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Beta :</string>
     </property>
    </widget>
    <widget class="QLabel" name="TauxAlpha">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>81</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Taux alpha :</string>
     </property>
    </widget>
    <widget class="QLabel" name="PeriodeAlpha">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>80</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Periode alpha :</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="TauxAlphaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>81</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.400000000000000</double>
     </property>
    </widget>
    <widget class="QSpinBox" name="PeriodeAlphaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>80</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>5</number>
     </property>
    </widget>
    <widget class="QLabel" name="TauxBeta">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Taux beta :</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="TauxBetaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>150</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.410000000000000</double>
     </property>
    </widget>
    <widget class="QSpinBox" name="PeriodeBetaValue">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>150</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>5</number>
     </property>
    </widget>
    <widget class="QLabel" name="PeriodeBeta">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>150</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Periode beta :</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QGraphicsView" name="graphicsView">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>0</y>
     <width>511</width>
     <height>591</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="PTUT.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>LigneValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>setRows()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ColValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>setColumns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Distance1</sender>
   <signal>clicked(bool)</signal>
   <receiver>Window</receiver>
   <slot>setEuclidian()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StartBtn</sender>
   <signal>clicked()</signal>
   <receiver>Window</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>525</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PauseBtn</sender>
   <signal>clicked()</signal>
   <receiver>Window</receiver>
   <slot>pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>525</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Titre1</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Line1</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>145</x>
     <y>127</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Lignes</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>35</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>LigneValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>90</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Colonnes</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>35</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>ColValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>90</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Titre2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>145</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Alpha</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>33</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Beta</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>33</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>BetaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>Distance1</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>StartBtn</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>535</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>AlphaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>247</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>TauxAlpha</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>45</x>
     <y>281</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>TauxAlphaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>281</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>PeriodeAlpha</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>PeriodeAlphaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>TauxBeta</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>45</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>TauxBetaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>PeriodeBeta</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>PeriodeBetaValue</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>350</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NouveauRes</sender>
   <signal>clicked(bool)</signal>
   <receiver>AlphaSlider</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>150</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TauxAlphaValue</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Window</receiver>
   <slot>alphaRateConstraint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TauxBetaValue</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Window</receiver>
   <slot>betaRateConstraint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AlphaSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Window</receiver>
   <slot>setAlphaValueText()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BetaValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>setBeta()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>320</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setRows()</slot>
  <slot>setColumns()</slot>
  <slot>setAlphaValueText()</slot>
  <slot>setBeta()</slot>
  <slot>setEuclidian()</slot>
  <slot>start()</slot>
  <slot>pause()</slot>
  <slot>alphaRateConstraint()</slot>
  <slot>betaRateConstraint()</slot>
 </slots>
</ui>
